### DISCUSSION TITLE
Can't reset file input

### DISCUSSION THREAD

**Discussion by morceaudebois** [livewire/livewire#8121] [1. Help] [answered]
Hi! I'm trying to clear a file input after submitting but I can't find how to make it work.

Here's what I have in my view:
```php
<div wire:click="resetFile()">Clear input</div>

<form wire:submit.prevent="addPlan" enctype="multipart/form-data">
    @csrf

    <input type="file" id="image" wire:model="image" name="image">
    <x-all.input-error :messages="$errors->get('image')" class="mt-2" />

    <x-all.primary-button type="submit">
        {{ __('Save') }}
    </x-all.primary-button>
</form>
```

And in my component:
```php
public $image;
public function resetFile() {
   $this->reset('image');
   // $this->image = null; // also tried this
}
```

The clear input button triggers but the file input doesn't reset, I'm probably doing something wrong but not sure what.

Thank you!

**Comment by danie-ramdhani** [comment]
you can't reset file input because it's not text.

maybe try delete the temporary uploaded file?

```php
// delete temporary files from livewire
$this->image->delete();
```

**Reply by morceaudebois** [reply]
Oh I see! `->delete()` seems to delete the value, but the the input field itself stays the same

**Reply by danie-ramdhani** [reply]
try:

```blade
<div wire:click="resetFile()" x-on:click="$el.closest('form').reset()">Clear input</div>
```

> [!NOTE]
> you need to wrap it inside form element

**Reply by morceaudebois** [reply]
This works! But the clear button situation was more for testing purposes, in reality I need to trigger the clearing after having validated and stored the data, so it wouldn't really work in that context.

I'm surprised Livewire doesn't seem to have a native way to deal with this

**Reply by danie-ramdhani** [reply]
you need to read this https://github.com/pi0/tired-maintainer or listen to this https://notesonwork.transistor.fm/episodes/no-you-fix-it

if you have a better solution you're welcome to make a PR 😅.

if it's done you can mark the answer / close the discussion.

**Reply by morceaudebois** [reply]
Didn't mean to be rude at all 😬

I was just thinking that it seems like a common situation, and in all the time Livewire has existed it seems likely that somebody had already thought of a solution. Thanks for the help!

**Reply by joshhanley** [reply]
@morceaudebois I haven't run into this before as typically when I save a form I then redirect to another page, hence not actually needing to clear the input. Maybe it's the same for other people?

Today's the first time I decided to upload a file in a modal, which then you need to reset the file input so the modal is ready for next use.

**Reply by morceaudebois** [reply]
@joshhanley Yeah, this is exactly what I'm doing too. I guess this just isn't that common

**Comment by joshhanley** [comment, marked as answer]
@morceaudebois I opened a PR #8120 just earlier today to fix this 🙂

**Reply by morceaudebois** [reply]
Ooh great! Thank you 😊

**Reply by joshhanley** [reply]
@morceaudebois No worries! 🙂
